### ISSUE TITLE
problems with current use of `cargo xtask verify-libraries`

### ISSUE THREAD

**Issue by iliana** [oxidecomputer/omicron#7149]
Currently we run `cargo xtask verify-libraries --release` as part of `cargo xtask releng`, and nowhere else in the CI process. There are two reasons this is probably wrong.

1. If you build a binary in dev mode (without `--release`) and copy it onto a intended-for-gimlet host OS image or into a zone, you should be able to run it as expected without any missing library issues. We are not testing this, and this in fact fails today (see below). Ideally I would like to determine the root cause of these new libraries when we're doing little-to-no LTO, either mark them as OK specifically for debug builds or eliminate them from debug builds somehow, and run `cargo xtask verify-binaries` in the clippy buildomat job. (I am not sure how critical this actually is.)
2. `cargo xtask verify-binaries --release` does not build binaries with the same exact feature flags that `omicron-package` does. Thus it is rebuilding binaries and checking the rebuilt ones after we've already packaged up the binaries into their relevant tarballs that get packaged up into the TUF repo. We are not actually checking the binaries that we are shipping, which was not my intention when moving this task into releng. I think I'd like to fix this by moving the binary-verification code out into a library and have releng run it directly on all of the binaries for packages we build.

cc @papertigers for your thoughts.

Here is the current output of `cargo xtask verify-libraries` (dev profile):

```
$ cargo xtask verify-libraries
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.15s
     Running `target/debug/xtask verify-libraries`
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 11.22s
Error: Found library issues with the following:
nexus
        UNEXPECTED dependency on libsqlite3.so.0
omicron-dev
        UNEXPECTED dependency on libsqlite3.so.0
zone-setup
        NEEDS libnvme.so.1 but is not allowed


If depending on a new library was intended please add it to xtask.toml
```

**Comment by papertigers** [comment]
```

**Comment by iliana** [comment]
libnvme-sys seems to get pulled into zone-setup via sled-agent, which uses it only to import a const:

```
$ rg sled_agent zone-setup/
zone-setup/src/bin/zone-setup.rs
21:use omicron_sled_agent::services::SWITCH_ZONE_BASEBOARD_FILE;
```

**Comment by papertigers** [comment]
Perhaps we can pull that into [sled-agent/types](https://github.com/oxidecomputer/omicron/tree/main/sled-agent/types) or `sled-hardware-types`?

**Comment by iliana** [comment]
Yeah, we should move that const into one of those crates. I'm honestly not sure where.

The nexus -> crucible-common dep was added in August, a few months after I moved the verify-libraries check into releng. I think this is a good candidate for a "dev-only" library; I think we clearly do not want Nexus relying on libsqlite for real, but I don't think it's worth the effort to excise it completely in the dep tree.
